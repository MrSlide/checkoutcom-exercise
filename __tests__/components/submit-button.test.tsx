import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { useFormContext } from 'react-hook-form'

import SubmitButton from '../../src/components/submit-button'

jest.mock('react-hook-form')

describe('SubmitButton component', function () {
  const mockUseFormContext = jest.mocked(useFormContext)

  test('renders correctly in idle state', function () {
    mockUseFormContext.mockReturnValue({
      formState: {
        isSubmitting: false,
        isValidating: false,
      },
    } as any)

    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <SubmitButton/>
    )

    expect(result).toMatchSnapshot()
  })

  test('renders correctly in form validating state', function () {
    mockUseFormContext.mockReturnValue({
      formState: {
        isSubmitting: false,
        isValidating: true,
      },
    } as any)

    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <SubmitButton/>
    )

    expect(result).toMatchSnapshot()
  })

  test('renders correctly in form submitting state', function () {
    mockUseFormContext.mockReturnValue({
      formState: {
        isSubmitting: true,
        isValidating: false,
      },
    } as any)

    const renderer = new ShallowRenderer()

    const result = renderer.render(
      <SubmitButton/>
    )

    expect(result).toMatchSnapshot()
  })
})
