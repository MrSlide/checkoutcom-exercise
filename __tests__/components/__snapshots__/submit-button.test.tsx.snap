// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton component renders correctly in form submitting state 1`] = `
<button
  aria-busy={true}
  className="secondary-button"
  disabled={true}
  type="submit"
/>
`;

exports[`SubmitButton component renders correctly in form validating state 1`] = `
<button
  aria-busy={true}
  className="secondary-button"
  disabled={true}
  type="submit"
/>
`;

exports[`SubmitButton component renders correctly in idle state 1`] = `
<button
  aria-busy={false}
  className="secondary-button"
  disabled={false}
  type="submit"
/>
`;
